pilot/pkg/model/gateway.go

```diff
@@ -73,19 +79,20 @@ func MergeGateways(gateways ...Config) *MergedGateway {
 							config.Name, s.Port.Name, s.Port.Number, s.Port.Protocol, p[0].Port.Name, p[0].Port.Number, p[0].Port.Protocol)
 						continue
 					}
-					rdsName := GatewayRDSRouteName(s)
-					if rdsName == "" {
+					routeName := gatewayRDSRouteName(s, config.Namespace)
```

Comment:
pass config as is.. use both name, namespace

Reply:
Yeah, I was hesitant to add gateway name right away -- tried to minimize changes of behavior until I get feedback on this from you or @costinm. Made the change. Thanks for review!